action/protocol/staking/staking_statereader.go

```diff
@@ -188,7 +188,7 @@ func (c *compositeStakingStateReader) readStateCandidates(ctx context.Context, r
 	)
 	if epochStartHeight != 0 && c.nativeIndexer != nil {
 		// read candidates from indexer
-		candidatesBytes, height, err = c.nativeIndexer.GetCandidates(height, req.GetPagination().GetOffset(), req.GetPagination().GetLimit())
+		candidatesBytes, height, err = c.nativeIndexer.GetCandidates(epochStartHeight, req.GetPagination().GetOffset(), req.GetPagination().GetLimit())
```

Comment:
change `nativeIndexer.GetCandidates` to directly return `CandidateList`, this can save a `Marshal` and `Unmarshal`

Reply:
in `candidate_buckets_indexer.go`